test/instances/ezETH/EzETHHyperdrive.t.sol

```diff
@@ -0,0 +1,1110 @@
+        uint256 expectedShares = getExpectedShares(
+            contribution,
+            totalPooledEtherBefore,
+            totalSharesBefore
+        );
+        assertApproxEqAbs(
+            hyperdrive.getPoolInfo().shareReserves,
+            expectedShares,
+            1
+        );
+        assertEq(
+            hyperdrive.getPoolInfo().lpTotalSupply,
+            hyperdrive.getPoolInfo().shareReserves - config.minimumShareReserves
+        );
+
+        // Verify that the correct events were emitted.
+        verifyFactoryEvents(
+            deployerCoordinator,
+            hyperdrive,
+            bob,
+            contribution,
+            FIXED_RATE,
+            true,
+            config.minimumShareReserves,
+            new bytes(0),
+            // NOTE: Tolerance since ezETH uses mulDivDown for share calculations.
+            1e5
+        );
+    }
+
+    function test__ezeth_interest_and_advance_time() external {
+        // hand calculated value sanity check
+        uint256 positionAdjustedInterestRate = uint256(0.05e18).mulDivDown(
+            POSITION_DURATION_2_WEEKS,
+            365 days
+        );
+
+        // Ensure that advancing time accrues interest like we expect.
+        (uint256 sharePriceBefore, , ) = getSharePrice();
+        advanceTime(POSITION_DURATION_2_WEEKS, 0.05e18);
+        (uint256 sharePriceAfter, , ) = getSharePrice();
+        assertEq(positionAdjustedInterestRate, 0.002054794520547945e18);
+        assertEq(
+            sharePriceBefore.mulDown(1e18 + positionAdjustedInterestRate),
+            sharePriceAfter
+        );
+    }
+
+    /// Price Per Share ///
+
+    function test__pricePerVaultShare(uint256 basePaid) external {
+        // Ensure that the share price is the expected value.
+
+        // Price in ETH / ezETH, does not include eigenlayer points.
+        (uint256 sharePrice, , ) = getSharePrice();
+        uint256 vaultSharePrice = hyperdrive.getPoolInfo().vaultSharePrice;
+        assertEq(vaultSharePrice, sharePrice);
+
+        // Ensure that the share price accurately predicts the amount of shares
+        // that will be minted for depositing a given amount of ETH.
+        basePaid = basePaid.normalizeToRange(
+            2 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            HyperdriveUtils.calculateMaxLong(hyperdrive)
+        );
+        uint256 hyperdriveSharesBefore = EZETH.balanceOf(address(hyperdrive));
+        openLong(bob, basePaid);
+
+        // NOTE: using the vaultSharePrice has somewhat higher tolerance.  If
+        // calculate with expectedShares(), we'll match exactly, but we're
+        // testing the vaultSharePrice here.
+        assertApproxEqAbs(
+            EZETH.balanceOf(address(hyperdrive)),
+            hyperdriveSharesBefore + basePaid.divDown(vaultSharePrice),
+            // TODO: verify tolerance ok, vaultSharePrice is pretty inaccurate
```

Comment:
Why is it any less accurate than Lido's share price?

Reply:
Are you asking me explain in the comment or here?  The reason AFAIK is that the vaultSharePrice here relys on converting one share to base, and that conversion doesn't account for the inflation like the base -> share conversion and also obviously becomes less accurate for larger conversions.

Reply:
I was asking to explain here, but my later comment supersedes this one. 

My reasoning for why the tolerances are so much higher for this yield source is that the minting calculation is the following:

This is their mint function:

```
        // For first mint, just return the new value added.
        // Checking both current value and existing supply to guard against gaming the initial mint
        if (_currentValueInProtocol == 0 || _existingEzETHSupply == 0) {
            return _newValueAdded; // value is priced in base units, so divide by scale factor
        }
        
        // Calculate the percentage of value after the deposit 
        uint256 inflationPercentaage = SCALE_FACTOR * _newValueAdded / (_currentValueInProtocol + _newValueAdded);

        // Calculate the new supply
        uint256 newEzETHSupply = (_existingEzETHSupply * SCALE_FACTOR) / (SCALE_FACTOR - inflationPercentaage);

        // Subtract the old supply from the new supply to get the amount to mint
        uint256 mintAmount = newEzETHSupply - _existingEzETHSupply;

        // Sanity check
        if(mintAmount == 0) revert InvalidTokenAmount();

        return mintAmount;
```

To analyze this a bit, this reduces to:

$$
\begin{aligned}
    i & = \frac{\Delta z}{z_0 + \Delta z} \\
    s_1 & = \frac{s_0}{1 - i}
\end{aligned}
$$

This looked weird to me, considering that they have a normal share price calculation when redeeming, but when I thought about it, I realized that it's doing the same thing as other yield sources:

$$
s_1 = \frac{s_0}{1 - i} = \frac{s_0}{1 - \tfrac{Delta z}{z_0 + \Delta z}} = \frac{s_0}{\tfrac{z_0}{z_0 + \Delta z}} = s_0 \cdot \frac{z_0 + \Delta z}{z_0}
$$

With this in mind, I think that what's happening is that by compressing the scaling factor into a value between 0 and 1, they lose a lot of precision compared to a normal share price calculation. This leads to an unavoidable loss in precision in our tests. Now that I think about it again this morning, I think this is fine from Hyperdrive's perspective since the imprecision doesn't effect most of Hyperdrive's accounting (Hyperdrive typically operates on the output of deposit, so it would receive the correct value). The one area where we could have issues is in `openShort`: https://github.com/delvtech/hyperdrive/blob/60c789d437ba4e77ce14495f931c45cabd1a643c/contracts/src/internal/HyperdriveShort.sol#L104. Instead of taking the output of `_deposit` as gospel, `openShort` assumes that the amount deposited is correct.

I think the lowest risk/easiest solution to these problems is to disable the ETH -> Renzo minting flow. We can still support this via a zap, but it wouldn't negatively impact Hyperdrive's precision.

Reply:
Ok so after our IRL conversation, I've disabled ETH -> ezETH minting flows.  We can reopen this in the future but taking the safe approach for now.